perf.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	AllCPUs  = "-a"
+	Output   = "-o"
+	Detailed = "-d"
+	Event    = "-e"
+	SEP      = "-x"
+)
+
+// PerfStat A instance of perf stat command
+type PerfStat struct {
+	args    string
+	sep     string
+	outFile string
+	run     chan bool
+	stop    chan bool
+}
+
+// NewPerfStat returns a new instance for perf stat
+func NewPerfStat(args ...string) *PerfStat {
+	perfStat := new(PerfStat)
+	perfStat.run = make(chan bool, 1)
+	perfStat.stop = make(chan bool, 1)
+	perfStat.args = ""
+	perfStat.sep = "|"
+
+	delimiter := ""
+	for i, arg := range args {
+		perfStat.args += delimiter + arg
+		delimiter = " "
+
+		switch arg {
+		case Output:
+			perfStat.outFile = args[i+1]
+		case SEP:
+			perfStat.sep = args[i+1]
+		}
+	}
+
+	if perfStat.sep == "" {
+		perfStat.sep = "|"
+		perfStat.args += delimiter + SEP + perfStat.sep
+	}
+
+	go func() {
+		for {
```

Comment:
I don't understand why you need this.

Reply:
Have changed the implementation of invoking perf.